engines/wintermute/base/base_object.cpp

```diff
@@ -1032,6 +1037,21 @@ bool BaseObject::persist(BasePersistenceManager *persistMgr) {
 	persistMgr->transferFloat(TMEMBER(_sFXParam3));
 	persistMgr->transferFloat(TMEMBER(_sFXParam4));
 
+#ifdef ENABLE_WME3D
+	if (BaseEngine::instance().getFlags() & GF_3D) {
```

Comment:
This has the downside that these variables will all be uninitialized after loading a game now. Perhaps outfactor the initialization part of the constructor to a new function that gets called both from the constructor and here, to make things slightly safer. Or, we could consider moving the 3d-related variables to a struct of their own that we instantiate in BaseObject, to keep things slightly easier to keep track of.

The reason this _may_ be dangerous, is that I assume that not ALL places that use these variables are guarded by the flag-check above, and thus they may need to have consistent values even in 2D games, just for safety.